Week1/homework/app.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+const root = document.getElementById("root");
+
+const hyfGithubRepositories = `https://api.github.com/orgs/HackYourFuture/repos?per_page=100`;
+
+
+function fetchJSON(callback, url) {
+    let xhr = new XMLHttpRequest();
+    xhr.open("GET", url, true);
+    xhr.onreadystatechange = () => {
+        if (xhr.readyState === 4 && xhr.status === 200) {
+            let response = JSON.parse(xhr.response);
+            callback(response);
+            return response;
+
+        } else if (xhr.status === 404) {
+            console.log("Error: resource not found");
+        }
+    }
+    xhr.send();
+}
+fetchJSON(storeResponse, hyfGithubRepositories);
+
+
+function storeResponse(xhrResponse) {
+    renderOptionTag(xhrResponse);
+    selectTag.addEventListener("change", e => renderData(e.target.value, xhrResponse));
+}
+
+function renderOptionTag(objects) {
+    objects.forEach(object => {
+        let optionTag = createAndAppend("option", selectTag);
+        optionTag.value = object.name;
+        optionTag.textContent = object.name;
+    });
+}
+
+let header = createAndAppend("header", root)
+let selectTag = createAndAppend("select", header);
+
+let wrapper = createAndAppend("div", root);
```

Comment:
While the code looks amazing, this should have been wrapped in a main() function.